swift/Arrow/Sources/Arrow/ArrowReaderHelper.swift

```diff
@@ -289,3 +289,10 @@ func validateFileData(_ data: Data) -> Bool {
     let endString = String(decoding: data[(data.count - markerLength)...], as: UTF8.self)
     return startString == FILEMARKER && endString == FILEMARKER
 }
+
+func getUInt32(_ data: Data, offset: Int) -> UInt32 {
+    let token = data.withUnsafeBytes { rawBuffer in
+        rawBuffer.loadUnaligned(fromByteOffset: offset, as: UInt32.self)
```

Comment:
Do we need a out-of-range offset check here?

Reply:
The method throws an error if the offset is out of range.  We can add a custom error but we use this same method in other places in the code and they do not throw a custom error.

Reply:
OK.